services/viva/api/cases/src/lambdas/addCasePerson.ts

```diff
@@ -0,0 +1,179 @@
+import { BadRequestError, ForbiddenError } from '@helsingborg-stad/npm-api-error-handling';
+
+import config from '../libs/config';
+import log from '../libs/logs';
+import { decodeToken, Token } from '../libs/token';
+import { populateFormWithPreviousCaseAnswers } from '../libs/formAnswers';
+import * as response from '../libs/response';
+import * as dynamoDb from '../libs/dynamoDb';
+
+import { cases } from '../helpers/query';
+import { getFormTemplates } from '../helpers/dynamoDb';
+import { VIVA_STATUS_NEW_APPLICATION_OPEN } from '../helpers/constants';
+import validateApplicationStatus from '../helpers/validateApplicationStatus';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+
+import { CasePersonRole } from '../types/caseItem';
+import type { CaseItem, CasePerson, CaseForm } from '../types/caseItem';
+import type { VivaApplicationStatus } from '../types/vivaMyPages';
+
+type CaseKeys = Pick<CaseItem, 'PK' | 'SK'>;
+
+interface AddCasePersonRequest {
+  personalNumber: string;
+  firstName: string;
+  lastName: string;
+}
+
+export interface LambdaRequest {
+  body: string;
+  pathParameters: {
+    caseId: string;
+  };
+  headers: {
+    Authorization: string;
+  };
+}
+
+interface LambdaResponse {
+  type: string;
+  attributes: {
+    caseItem: CaseItem;
+  };
+}
+
+interface UpdateCaseAddPersonResponse {
+  Attributes: CaseItem;
+}
+
+interface UpdateCaseParameters {
+  caseKeys: CaseKeys;
+  coApplicant: CasePerson;
+  form: Record<string, CaseForm>;
+}
+
+export interface Dependencies {
+  decodeToken: (params: LambdaRequest) => Token;
+  updateCase: (params: UpdateCaseParameters) => Promise<UpdateCaseAddPersonResponse>;
+  coApplicantStatus: (personalNumber: string) => Promise<unknown>;
+  validateCoApplicantStatus: typeof validateApplicationStatus;
+  getCase: (keys: CaseKeys) => Promise<CaseItem>;
+  getFormTemplates: typeof getFormTemplates;
+  populateForm: typeof populateFormWithPreviousCaseAnswers;
```

Comment:
Is it to much work to add correct typings for `populateFormWithPreviousCaseAnswers` in `formAnswers.d.ts` ?

Reply:
some

Reply:
Will be fixed in the future :)